packages/flutter/test/gestures/pointer_data_resampler_test.dart

```diff
@@ -0,0 +1,443 @@
+    expect(result[2].change, PointerChange.up);
+    expect(result[2].physicalX, 25.0);
+    expect(result[2].physicalY, 0.0);
+    expect(result[2].physicalDeltaX, 0.0);
+    expect(result[2].physicalDeltaY, 0.0);
+    expect(result[3].timeStamp, const Duration(microseconds: 4500));
+    expect(result[3].change, PointerChange.remove);
+    expect(result[3].physicalX, 25.0);
+    expect(result[3].physicalY, 0.0);
+    expect(result[3].physicalDeltaX, 0.0);
+    expect(result[3].physicalDeltaY, 0.0);
+
+    result = resampler.sample(const Duration(microseconds: 5500));
+
+    // No pointer data should have been returned.
+    expect(result.isEmpty, true);
+  });
+
+  test('skip all', () {
+    final PointerDataResampler resampler = PointerDataResampler();
+    final PointerData data0 = _createSimulatedPointerData(
+        PointerChange.add, 1000, 0.0, 0.0, 0.0, 0.0);
+    final PointerData data1 = _createSimulatedPointerData(
+        PointerChange.down, 1000, 0.0, 0.0, 0.0, 0.0);
+    final PointerData data2 = _createSimulatedPointerData(
+        PointerChange.up, 4000, 30.0, 0.0, 30.0, 0.0);
+    final PointerData data3 = _createSimulatedPointerData(
+        PointerChange.remove, 4000, 30.0, 0.0, 0.0, 0.0);
+
+    resampler..addData(data0)..addData(data1)..addData(data2)..addData(data3);
+
+    List<PointerData> result =
+        resampler.sample(const Duration(microseconds: 500));
+
+    // No pointer data should have been returned.
+    expect(result.isEmpty, true);
+
+    result = resampler.sample(const Duration(microseconds: 5500));
+
+    // All remaining pointer data should have been returned.
+    expect(result.length, 4);
+    expect(result[0].timeStamp, const Duration(microseconds: 5500));
+    expect(result[0].change, PointerChange.add);
+    expect(result[0].physicalX, 30.0);
+    expect(result[0].physicalY, 0.0);
+    expect(result[0].physicalDeltaX, 0.0);
+    expect(result[0].physicalDeltaY, 0.0);
+    expect(result[1].timeStamp, const Duration(microseconds: 5500));
+    expect(result[1].change, PointerChange.down);
+    expect(result[1].physicalX, 30.0);
+    expect(result[1].physicalY, 0.0);
+    expect(result[1].physicalDeltaX, 0.0);
+    expect(result[1].physicalDeltaY, 0.0);
+    expect(result[2].timeStamp, const Duration(microseconds: 5500));
+    expect(result[2].change, PointerChange.up);
+    expect(result[2].physicalX, 30.0);
+    expect(result[2].physicalY, 0.0);
+    expect(result[2].physicalDeltaX, 0.0);
+    expect(result[2].physicalDeltaY, 0.0);
+    expect(result[3].timeStamp, const Duration(microseconds: 5500));
+    expect(result[3].change, PointerChange.remove);
+    expect(result[3].physicalX, 30.0);
+    expect(result[3].physicalY, 0.0);
+    expect(result[3].physicalDeltaX, 0.0);
+    expect(result[3].physicalDeltaY, 0.0);
+
+    result = resampler.sample(const Duration(microseconds: 6500));
+
+    // No pointer data should have been returned.
+    expect(result.isEmpty, true);
+  });
```

Comment:
All the above unit tests add all events at once, and then sample gradually later. In practice, I'd expect most cases to add a few events, sample some, and then adding a few more, and then sample again. Shall we also unit test such cases?

Reply:
Done. Latest code as a new "stream" test that behaves more like what we see on real devices.